### DISCUSSION TITLE
How to select device during initialization

### DISCUSSION THREAD

**Discussion by pinczakko** [Tencent/ncnn#2705] [General]
Hi all,
I read through ncnn source code and still a bit confused on how to enable choosing particular GPU device. I've found the _create_gpu_instance()_ and _find_default_vulkan_device_index()_ functions in **src/gpu.cpp**, but have yet to find where the former is called from. 
Anyway, what I want to do is to benchmark ncnn inference in a GPU which is _not_ the default GPU. I ran the inference (yolov4) in a laptop with Vega 10 integrated GPU and an Nvidia discrete GPU, but ncnn always runs on the Vega 10 integrated GPU.  I'm running the inference test in Windows 10, btw.
In _find_default_vulkan_device_index()_, the code comment suggests it will try the discrete GPU first, but it didn't workout that way. 
This is the screenshot when testing the example code:
![Screenshot (171)](https://user-images.githubusercontent.com/11014633/109008486-f8452e00-76df-11eb-850a-b2d635edb86b.png)

Any suggestion?

Thanks in advance

**Comment by nihui** [comment]
```
net.set_vulkan_device(1);
```

before loading param and model

**Reply by pinczakko** [reply]
ok, thanks. It works.
